engines/private/private.cpp

```diff
@@ -0,0 +1,1086 @@
+        stream->writeByte(0);
+    }
+
+    // Radios
+    stream->writeUint32LE(_AMRadio.size());
+    for (SoundList::iterator it = _AMRadio.begin(); it != _AMRadio.end(); ++it) {
+        stream->writeString(*it);
+        stream->writeByte(0);
+    }
+    stream->writeUint32LE(_policeRadio.size());
+    for (SoundList::iterator it = _policeRadio.begin(); it != _policeRadio.end(); ++it) {
+        stream->writeString(*it);
+        stream->writeByte(0);
+    }
+
+    stream->writeUint32LE(_phone.size());
+    for (PhoneList::iterator it = _phone.begin(); it != _phone.end(); ++it) {
+        stream->writeString(*it->sound);
+        stream->writeByte(0);
+        stream->writeString(*it->flag->name);
+        stream->writeByte(0);
+        stream->writeUint32LE(it->val);
+    }
+
+    // Played media
+    stream->writeString(_repeatedMovieExit);
+    stream->writeByte(0);
+
+    stream->writeUint32LE(_playedMovies.size());
+    for (PlayedMediaTable::iterator it = _playedMovies.begin(); it != _playedMovies.end(); ++it) {
+        stream->writeString(it->_key);
+        stream->writeByte(0);
+    }
+
+    stream->writeUint32LE(_playedPhoneClips.size());
+    for (PlayedMediaTable::iterator it = _playedPhoneClips.begin(); it != _playedPhoneClips.end(); ++it) {
+        stream->writeString(it->_key);
+        stream->writeByte(0);
+    }
+
+    return Common::kNoError;
+}
+
+Common::String PrivateEngine::convertPath(Common::String name) {
+    Common::String path(name);
+    Common::String s1("\\");
+    Common::String s2("/");
+
+    while (path.contains(s1))
+        Common::replace(path, s1, s2);
+
+    s1 = Common::String("\"");
+    s2 = Common::String("");
+
+    Common::replace(path, s1, s2);
+    Common::replace(path, s1, s2);
+
+    path.toLowercase();
+    return path;
+}
+
+void PrivateEngine::playSound(const Common::String &name, uint loops, bool stopOthers, bool background) {
+    debugC(1, kPrivateDebugFunction, "%s(%s,%d,%d,%d)", __FUNCTION__, name.c_str(), loops, stopOthers, background);
+
+    Common::File *file = new Common::File();
+    Common::String path = convertPath(name);
+
+    if (!file->open(path))
+        error("unable to find sound file %s", path.c_str());
+
+    Audio::LoopingAudioStream *stream;
+    stream = new Audio::LoopingAudioStream(Audio::makeWAVStream(file, DisposeAfterUse::YES), loops);
+    if (stopOthers) {
+        stopSound(true);
+    }
+
+    Audio::SoundHandle *sh = NULL;
+    if (background) {
+        _mixer->stopHandle(_bgSoundHandle);
+        sh = &_bgSoundHandle;
+    } else {
+        _mixer->stopHandle(_fgSoundHandle);
+        sh = &_fgSoundHandle;
+    }
+
+    _mixer->playStream(Audio::Mixer::kSFXSoundType, sh, stream, -1, Audio::Mixer::kMaxChannelVolume);
+}
+
+void PrivateEngine::playVideo(const Common::String &name) {
+    debugC(1, kPrivateDebugFunction, "%s(%s)", __FUNCTION__, name.c_str());
+    //stopSound(true);
+    Common::File *file = new Common::File();
```

Comment:
You're leaking it.

Reply:
It seems I'm not leaking this File pointer. It is automatically deleted once this call is complete:

```cpp
    stream = new Audio::LoopingAudioStream(Audio::makeWAVStream(file, DisposeAfterUse::YES), loops);
```

Attempting to free the `file`  pointer results in a double free.